scrapy_dropbox/utils.py

```diff
@@ -0,0 +1,15 @@
+import os
+
+
+def parse_uri(uri):
+    """Takes in uri and returns the dropbox path"""
+    if not uri.startswith("dropbox://"):
+        return None
+
+    dropbox_path = uri.replace("dropbox:/", "")
+    _, file_extension = os.path.splitext(dropbox_path)
+
+    if not file_extension:
+        return None
+
+    return dropbox_path
```

Comment:
I think the old implementation was wrong in hardcoding `dbox`, and this one is wrong in hard-coding `dropbox`.

What would you think about accepting whatever schema the user chose (dropbox, dbox, d, x), and ignore it? e.g. `uri.replace("://", "/", 1)`.

Also, why are we failing on the lack of a file extension? It should be OK to upload a file without an extension to Dropbox, right?

Reply:
> I think the old implementation was wrong in hardcoding dbox, and this one is wrong in hard-coding dropbox.

Thanks @Gallaecio . 
The uri scheme in the "FEEDS" dict needs to match with the feed storage name in the FEED_STORAGES dict. Otherwise our feed storage won't be detected/triggered.
e.g: 
```
    custom_settings = {
        "DROPBOX_API_TOKEN": "xxxxx",
        "FEED_STORAGES": {
            'dropbox': 'test_dropbox.DropboxFeedStorage'
        },
        "FEEDS":{
            "dropbox://testfile": {
                "format": "json",
                "encoding": "utf8"
            }
        },
    }
```

That's why I've hardcoded "dropbox"

> Also, why are we failing on the lack of a file extension? It should be OK to upload a file without an extension to Dropbox, right?

Yes, I agree with that so I've made the change. Thank you.

Reply:
> That's why I've hardcoded "dropbox"

My point is that users are free to configure FEED_STORAGES however they like, i.e. some users may define:

```
        "FEED_STORAGES": {
            'dropbox': 'test_dropbox.DropboxFeedStorage'
        },
```

Some may prefer shorter URLs:

```
        "FEED_STORAGES": {
            'dbox': 'test_dropbox.DropboxFeedStorage'
        },
```

Some may use a generic or internal URL prefix:

```
        "FEED_STORAGES": {
            'internal-storage': 'test_dropbox.DropboxFeedStorage'
        },
```

And some may support multiple syntaxes for the same storage, who knows.

```
        "FEED_STORAGES": {
            'dropbox': 'test_dropbox.DropboxFeedStorage',
            'dbox': 'test_dropbox.DropboxFeedStorage',
        },
```

And we can make all those scenarios work. We do not need to force DropBox URLs to use `dropbox` as URL scheme, even if we encourage and recommend that. And this way, we maintain backward-compatibility.

Reply:
Done :+1: